spanner/benchmark/ycsb.py

```diff
@@ -0,0 +1,259 @@
+"""The YCSB cient in Python.
+
+Usage:
+
+  # Set up instance and load data into database.
+
+  # Set up environment variables. You should use your own credentials and gclod
+  # project.
+  $ export GOOGLE_APPLICATION_CREDENTIALS=/path/to/credentials.json
+  $ export GCLOUD_PROJECT=gcloud-project-name
+
+  # Run the benchmark.
+  $ python spanner/benchmark/ycsb.py run -P pkb/workloada -p table=usertable \
+    -p cloudspanner.instance=ycsb-542756a4 -p recordcount=5000 \
+    -p operationcount=100 -p cloudspanner.database=ycsb -p num_worker=1
+
+  # To make a package so it can work with PerfKitBenchmarker.
+  $ cd spanner; tar -cvzf ycsb-python.0.0.5.tar.gz benchmark/*
+
+"""
+
+import numpy
+import optparse
+import random
+import string
+import threading
+import timeit
+
+from google.cloud import spanner
+
+
+KEYS = []
+OPERATIONS = ['readproportion', 'updateproportion', 'scanproportion',
+              'insertproportion']
+
+
+def ParseOptions():
+  """Parses options."""
+  parser = optparse.OptionParser()
+  parser.add_option('-P', '--workload', action='store', dest='workload',
+                    default='', help='The path to a YCSB workload file.')
+  parser.add_option('-p', '--parameter', action='append', dest='parameters',
+                    default=[], help='The key=value pair of parameter.')
+  parser.add_option('-b', '--num_bucket', action='store', type='int',
+                    dest='num_bucket', default=1000,
+                    help='The number of buckets in output.')
+
+  options, args = parser.parse_args()
+
+  parameters = {}
+  parameters['command'] = args[0]
+  parameters['num_bucket'] = options.num_bucket
+
+  for parameter in options.parameters:
+    parts = parameter.strip().split('=')
+    parameters[parts[0]] = parts[1]
+
+  with open(options.workload, 'r') as f:
+    for line in f.readlines():
+      parts = line.split('=')
+      key = parts[0].strip()
+      if key in OPERATIONS:
+        parameters[key] = parts[1].strip()
+
+  return parameters
+
+
+def OpenDatabase(parameters):
+  """Opens a database specified by the parameters from ParseOptions()."""
+  spanner_client = spanner.Client()
+  instance_id = parameters['cloudspanner.instance']
+  instance = spanner_client.instance(instance_id)
+  database_id = parameters['cloudspanner.database']
+  pool = spanner.BurstyPool(int(parameters['num_worker']))
+  database = instance.database(database_id, pool=pool)
+
+  return database
+
+
+def LoadKeys(database, parameters):
+  """Loads keys from database."""
+  global KEYS
+  KEYS = []
```

Comment:
Would it be better to use KEYS.clear() here so you don't need global?  Then you wouldn't be reassigning.